aws/session/internal/ini/visitor.go

```diff
@@ -0,0 +1,135 @@
+package ini
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// Visitor is an interface used by walkers that will
+// traverse an array of ASTs.
+type Visitor interface {
+	VisitExpr(AST) error
+	VisitStatement(AST) error
+}
+
+// Sections is a map of Section structures that represent
+// a configuration.
+type Sections map[string]Section
+
+// SharedConfigVisitor is used to visit statements and expressions
+// and ensure that they are both of the correct format.
+// In addition, upon visiting this will build sections and populate
+// the Sections field which can be used to retrieve profile
+// configuration.
+type SharedConfigVisitor struct {
```

Comment:
Wouldn't refer to `SharedConfig` here something more like `Default` would be better